### ISSUE TITLE
Remove scikit-learn dependency

### ISSUE THREAD

**Issue by rain1024** [undertheseanlp/underthesea#919]
## Motivation

`scikit-learn>=1.6.1` is a heavy dependency (~30MB) that brings in numpy, scipy, and other packages. For a lightweight NLP toolkit, this adds significant installation time and disk space.

**Goal: Remove sklearn from inference path.** Training can still use sklearn as an optional dependency.

### 1. Pre-trained Models (Main Blocker)
- Bank classifier (`sonar_core_1`) is a sklearn Pipeline saved with joblib
- Models use `predict_proba()`, `classes_` attributes
- Location: `underthesea/pipeline/classification/bank/__init__.py`

### 2. Text Classifier (`underthesea/models/text_classifier.py`)
- `MultiLabelBinarizer` for multi-label prediction decoding
- sklearn Pipeline loading and prediction

### 3. Custom Transformers (used in saved models)
- `underthesea/pipeline/classification/text_features.py`
- `underthesea/pipeline/sentiment/bank/text_features.py`
- `underthesea/pipeline/sentiment/general/text_features.py`
- These inherit from `BaseEstimator`, `TransformerMixin`

## Out of Scope (Training Only)

These can remain with sklearn in `[train]` optional dependency:
- `CountVectorizer`, `TfidfVectorizer` in `underthesea/transformer/`
- `LabelEncoder`, `f1_score` in `underthesea/trainers/`

### Step 1: Move sklearn to optional dependency
```toml
# pyproject.toml
dependencies = [
    # Remove scikit-learn from here
]

[project.optional-dependencies]
train = [
    "scikit-learn>=1.6.1",
    "seqeval"
]
```

### Step 2: Re-train models with sklearn-free format
- Convert bank classifier to use custom vectorizer + simple classifier
- Save model in format that doesn't require sklearn to load
- Options:
  - Pure Python dict-based model
  - ONNX format
  - Custom binary format via `underthesea_core` (Rust)

### Step 3: Update inference code
| Current | Replacement |
|---------|-------------|
| sklearn Pipeline | Custom pipeline class |
| `MultiLabelBinarizer` | Simple list-based decoder |
| `BaseEstimator` | Remove inheritance |
| `predict_proba()` | Custom predict method |

## Tasks

- [ ] Audit all inference paths that import sklearn
- [ ] Design sklearn-free model format
- [ ] Re-train bank classifier without sklearn
- [ ] Update `TextClassifier.load()` and `predict()`
- [ ] Move sklearn to `[train]` optional dependency
- [ ] Update tests

## Related Issues

- #731 - sklearn 1.5+ compatibility issues with `check_is_fitted`

**Comment by rain1024** [comment]
Hi @BLKSerene,

Thank you for bringing this up and for linking the related issue #586.

I'm currently working on retraining the models using a new Rust-based implementation (`underthesea_core_extend`). The initial benchmarks are very promising — the Rust inference path is significantly faster than the sklearn version, and more importantly, it will allow us to completely remove the sklearn dependency from the inference path.

This is actively in progress and should be resolved soon. I'll update this issue once the new models are ready.

Thanks for your patience!

**Comment by rain1024** [comment]
All tasks completed and merged:

- **PR #946** — Retrained sentiment models with `underthesea_core.TextClassifier` (Rust), rewrote `sentiment/general/__init__.py` and `sentiment/bank/__init__.py` to use underthesea_core, updated classification model URLs, updated tests
- **PR #952** — Removed `scikit-learn` from `pyproject.toml` dependencies, deleted 7 legacy files (839 lines): `models/text_classifier.py`, `trainers/classifier_trainer.py`, `transformer/tfidf.py`, `transformer/count.py`, and 3 `text_features.py` files

`joblib` is kept for CRF tagger and text normalization serialization.
